### ISSUE TITLE
Window Buttons break after maximizing/unmaximizing

### ISSUE THREAD

**Issue by aragubas** [kupiqu/SierraBreezeEnhanced#116]
Window buttons (close, minimize, maximize) breaks after maximizing/unmaxizing (restoring)

Plasma Version: 5.27.3
KWin Version: 5.27.3
Compiled using the 1.3.1 release

![Screenshot_20230318_112737](https://user-images.githubusercontent.com/47148297/226112215-f43c7022-2ab0-4222-a7f9-f7e043a128a7.png)
![Screenshot_20230318_112752](https://user-images.githubusercontent.com/47148297/226112217-b92398c7-c86f-4ba1-847e-ac6e1f00d4e8.png)

**Comment by aragubas** [comment]
Still broken when compiling with the latest branch (git clone)

![image](https://user-images.githubusercontent.com/47148297/226112490-ab8e9692-81fc-411e-89da-36fdab8d3c90.png)

**Comment by aragubas** [comment]
Doesn't happen when the buttons are on the right, only happens when the buttons are on the left

**Comment by aragubas** [comment]
Still happens when changing the order of the buttons, so its a problem when rendering the buttons on the left

![image](https://user-images.githubusercontent.com/47148297/226112710-7a28090f-55e0-4043-9b70-2e77f9d67b3b.png)

**Comment by kupiqu** [comment]
I cannot reproduce. It is probably a duplicate of issue #111

**Comment by kupiqu** [comment]
I finally could reproduce it. Duplicate of #111, Let's keep discussion there

**Comment by kupiqu** [comment]
> Awesome! Out of curiosity, did you manage to find out why you couldn't reproduce it at first?

To be frank, I'm not at all sure. Last time I tried before, I tested a bunch of options and nothing triggered the issue. Today it started the same. At some point I left a single icon on the right and then the issue appeared. As I played with many things in parallel I am not 100% confident that this was the actual trigger. In fact, nothing in the fix is related to the number of icons on the rightside.

Perhaps it is actually a bug elsewhere or in Kwin/Qt since to my understanding there shouldn't be a difference between the two options here:

![image](https://user-images.githubusercontent.com/3875196/226317604-b6fadac4-a849-4c97-ad9a-8a3ad04fd050.png)

Even less sth to trigger only after restoring from maximization (which ends up duplicating the horizontal offset somehow)
